internal/metrics/metrics.go

```diff
@@ -127,3 +129,44 @@ func (s *ScheduleTimer) Mark(ctx context.Context, role, outcome string) {
 	metrics.FromContext(ctx).Histogram(
 		schedulingRetryTime, sparkRoleTag, outcomeTag, s.instanceGroupTag, s.retryTag).Update(now.Sub(s.lastSeenTime).Nanoseconds())
 }
+
+// ReportCrossZoneMetric reports metric about cross AZ traffic between pods of a spark application
+func ReportCrossZoneMetric(ctx context.Context, driverNodeName string, executorNodeNames []string, nodes []*v1.Node) {
+	executorNodesSet := make(map[string]bool)
```

Comment:
you should use a struct{} here instead of a bool

Reply:
we already have this convention of using this type for "sets" in scheduler

Reply:
ok, but just flagging that this isn't idomatic go :)

Reply:
I think k8s codebase uses `bool`s for sets

Reply:
https://github.com/kubernetes/kubernetes/tree/v1.12.4/staging/src/k8s.io/apimachinery/pkg/util/sets

That uses struct{}. It's fine, just a matter a few bytes :D

Reply:
I don't know which one is the idiomatic way though. I see your example but there are far more bool sets in k8s/src then `struct{}`s
I agree `struct{}` has a better memory footprint, but if we care enough to change this, I think this should be a separate PR changing all set representations in the codebase. Happy to merge that though.